niap-cc/Permissions/Tester/app/src/main/java/com/android/certifications/niap/permissions/SignaturePermissionTester.java

```diff
@@ -3036,18 +3217,30 @@ public Object invoke(Object o, Method method, Object[] objects)
                 }));
 
         mPermissionTasks.put(permission.CONTROL_DEVICE_STATE,
-                new PermissionTest(false, Build.VERSION_CODES.S, () -> {
-                    Object deviceStateManager = mContext.getSystemService("device_state");
-                    Class deviceStateRequestClass = null;
+                new PermissionTest(false, Build.VERSION_CODES.S,
+                        Build.VERSION_CODES.S, () -> {
+                    @SuppressLint("WrongConstant") Object deviceStateManager
+                            = mContext.getSystemService("device_state");
+                    Class<?> deviceStateRequestClass = null;
                     try {
                         deviceStateRequestClass = Class.forName(
                                 "android.hardware.devicestate.DeviceStateRequest");
                     } catch (ClassNotFoundException e) {
                         throw new UnexpectedPermissionTestFailureException(e);
                     }
-                    mTransacts.invokeTransact(Transacts.DEVICE_STATE_SERVICE,
-                            Transacts.DEVICE_STATE_DESCRIPTOR, Transacts.cancelRequest,
-                            deviceStateRequestClass.cast(null));
+                    if (mDeviceApiLevel <= Build.VERSION_CODES.S) {
```

Comment:
Nit: is this conditional and the else block necessary? If the min and max SDK versions for this test are S, shouldn't this only run on devices running Android S and the else block will never be executed?

Reply:
Okay. Removed the check.

Reply:
Change looks good, thanks!